lib/azure/service_management/role.rb

```diff
@@ -548,5 +562,144 @@ def create(params, roleXML)
       "/#{params['deploy_name']}/roles"
       @connection.query_azure(servicecall, "post", roleXML.to_xml)
     end
+
+    def setup_extension(params)
+      role_xml = update_role_xml(params[:role_xml], params) ## add Chef Extension config in VM's role_xml
+
+      ## create new XML (with Chef Extension config and other pre-existing VM config) using the role_xml retrieved from the VM
```

Comment:
Please add comment specifying why we can't use the `role_xml` retrieved above as it is and have to create a New XML